src/graphics/render_stl.cpp

```diff
@@ -106,42 +107,45 @@ class Transformation_matrix
 class Stl_context
 {
 private:
-	FILE *stl_file;
-	char *solid_name;
+    std::string stl_content;
+    std::string solid_name;
 	/* Future: add binary option */
 	std::stack<Transformation_matrix> transformation_stack;
 
 public:
-	Stl_context(const char *file_name, const char *solid_name_in) :
-		stl_file(fopen(file_name, "w"))
+    Stl_context(const char *solid_name_in) :
+        stl_content(),
+        solid_name()
 	{
 		/* ASCII STL header */
 		if (solid_name_in == 0)
-			solid_name = duplicate_string("default");
+            solid_name = "default";
 		else
-			solid_name = duplicate_string(solid_name_in);
+            solid_name = std::string(solid_name_in);
 
-		fprintf(stl_file,"solid %s\n",solid_name);
+        stl_content += "solid " + solid_name + "\n";
 	}
 
 	~Stl_context()
 	{
-		fprintf(stl_file,"endsolid %s\n",solid_name);
-		DEALLOCATE(solid_name);
-		fclose(stl_file);
 	}
 
-	/***************************************************************************//**
+    std::string get_export_string() const
+    {
+        return stl_content + "endsolid " + solid_name + "\n";
+    }
+
+    /**
 	 * Confirms STL file is correctly opened.
 	 * 
 	 * @return true if context is correctly constructed, false if not.
 	 */
 	bool is_valid() const

```

Comment:
Is this separate verification needed now? Can we assume the constructor, or a possible begin solid method will throw and exception if not OK?
If keeping, update comment as not a file.